[herd] Fixes for -variant vmsa,mte

relokin opened the pull request:
This PR fixes three different issues in herd7 when simulating with `-variant vmsa,mte`.
1. Tag locations in post-condition are Physical symbols when running with `-variant vmsa`.
2. Fixes the possible executions for `STZ{,2}G` when there is a fault.
3. Fixes `lift_kvm` to correctly handle cases where a checked instruction runs into a fault.

For the 3 fixes, this PR contributes unit tests that demonstrate the issue and serve as regressions after the fix is applied.

relokin commented:
@maranget from my perspective this PR is now ready and the instruction semantics have been reviewed and agreed. Would you like to have a look? If you do, there are some motivating unit tests but if there is something that doesn't make sense please let me know.

maranget commented:
LGTM. I agree that the PR can be merged, once rebased on master, as usual.

As far as I have checked the output and diagrams of tests A010, A011 and A012 has not changed from before the PR: a small contraction with commit  bfae4ce title "Fix fault behaviour for -variant vmsa,mte". Or am I missing something?

relokin commented:
> LGTM. I agree that the PR can be merged, once rebased on master, as usual.
> 
> As far as I have checked the output and diagrams of tests A010, A011 and A012 has not changed from before the PR: a small contraction with commit [bfae4ce](https://github.com/herd/herdtools7/commit/bfae4ceb8ce726b94fa5d82d1a9ce1bd3384c143) title "Fix fault behaviour for -variant vmsa,mte". Or am I missing something?

Good point, these are interesting tests illustrative of the semantics that we want to model, but they didn't change due to this commit. I've added 3 more tests that would fail before this fix.

Also rebased to the latest master.

maranget commented:
Merged, thanks @relokin.

relokin commented:
Super! Many thanks @maranget!
